### PR TITLE
VACMS 16148 update phone numbers

### PR DESCRIPTION
**Pull request by eselkin**
## Summary

- Changes `<a>` telephone links to `<va-telephone>` components. Updates filters for parsing phone numbers. Updates phone templates as well. Updates VAMC, Vet Center, and pre-release VBA pages
- Sitewide Facilities

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-cms/issues/16148

## Testing done

- Manually checked that `<va-telephone>` appear and have correct aria-describedby when appropriate
- Automated testing is handled more in the vets-website PR
- Added tests for content-build filter with fallback from `<va-telephone>` to `<a href="tel:+1">`.

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

<details>
<summary>
VAMC system page
</summary>

</details>

<details>
<summary>
VAMC Locations Listing Page ("Other Nearby Locations" is in a PR to vets-website)
</summary>

</details>

<details>
<summary>
VAMC Billing and Insurance
</summary>

</details>

<details>
<summary>
VAMC Facilities
</summary>

</details>

<details>
<summary>
VAMC Leadership page
</summary>

</details>

<details>
<summary>
VAMC Leadership Detail Page (Bio)
</summary>

</details>

<details>
<summary>VAMC Region Detail Page 
</summary>

</details>

## What areas of the site does it impact?

Affects phone numbers on all VAMC pages. (All Vet Center and VBA changes that can be changed are on vets-website PR)

## Acceptance criteria

- [x] VAMC:
  - [x] All phone numbers on System template are updated to use the component with the `message-aria-describedby` prop
  - [x] Phone numbers on the Locations template have been updated to use the component with the `message-aria-describedby` prop
  - [x] On the Locations template, "Other nearby VA locations" have been updated with the `message-aria-describedby` prop
  - [x] Phone number on Billing and Insurance has been updated to use the component
  - [x] Phone numbers on the Facilities template have been updated to use the component with the `message-aria-describedby` prop
  - [x] Staff Listing template has been updated to use the component
  - [x] Staff Detail template has been updated to use the component
  - [x] Staff Profile phone number has been updated to use the component
- [ ] Phone number in the Vet center alert has been updated to use the component (comes in as a phone anchor tag from drupal)
- [x] VBA Benefits Hotline number has been updated with `message-aria-describedby` prop
- [ ] Both phone numbers in VBA alert have been updated to use the component (come in as anchor tags)
- [x] Exts in all phone numbers are parsed correctly
- [ ] Templates indicated as having CMS content only have been reviewed to ensure nothing has been missed
- [ ] component is wrapped in `<p class="vads-u-margin-top--0">` vs a generic `<div>` (not going to do because most are inline and cannot determine in phoneLinks replace links when a phone number is on its own line. We could house them in `<span>` or something non-block-like. All instances when a phone number is on its own line in a section are in a `<p>` already.)
- [x] Update and run existing test(s).
- [x] If additional tests are needed, stub a ticket to note this for future work.
- [x] Updates are validated across [VA Supported browsers](https://depo-platform-documentation.scrollhelp.site/developer-docs/browser-support-policy)
- [x] Requires accessibility review (Laura is tagged for PR review)

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user (nothing affecting state or authentication)

## Requested Feedback

Top 2 numbers on:
https://web-ihqrv6rby5lxlkn8zqdxf9u7lk1aplcp.demo.cms.va.gov/cleveland-va-regional-benefit-office/

Top number and spotlights (when link not sent from CMS)
https://web-ihqrv6rby5lxlkn8zqdxf9u7lk1aplcp.demo.cms.va.gov/boston-vet-center/

VAMC system
https://web-ihqrv6rby5lxlkn8zqdxf9u7lk1aplcp.demo.cms.va.gov/boston-health-care/

VAMC Facility
https://web-ihqrv6rby5lxlkn8zqdxf9u7lk1aplcp.demo.cms.va.gov/boston-health-care/locations/jamaica-plain-va-medical-center/

Register for Care
https://web-ihqrv6rby5lxlkn8zqdxf9u7lk1aplcp.demo.cms.va.gov/western-colorado-health-care/register-for-care/

Billing and Insurance
https://web-ihqrv6rby5lxlkn8zqdxf9u7lk1aplcp.demo.cms.va.gov/western-colorado-health-care/billing-and-insurance/

Leadership
https://web-ihqrv6rby5lxlkn8zqdxf9u7lk1aplcp.demo.cms.va.gov/western-colorado-health-care/about-us/leadership/

Bio
https://web-ihqrv6rby5lxlkn8zqdxf9u7lk1aplcp.demo.cms.va.gov/western-colorado-health-care/staff-profiles/richard-salgueiro/

Detail page with bio
https://web-ihqrv6rby5lxlkn8zqdxf9u7lk1aplcp.demo.cms.va.gov/boston-health-care/work-with-us/jobs-and-careers/nursing-careers/

### CODE REVIEW

**Comment by eselkin** [comment]
The code is ready for review, but the Tugboat's still a couple of hours off due to multiple release requests.

File: src/site/components/phone.drupal.liquid [hunk 1]
```diff
@@ -8,7 +8,10 @@
           phoneExtension = number.fieldPhoneExtension
           phoneLabel = number.fieldPhoneLabel | default: 'Phone'
           phoneHeaderLevel = phoneHeaderLevel
-        %}
+          fax = false
+          sms = false
+          tty = false
```
Comment (maxx1128) [inline src/site/components/phone.drupal.liquid:13]: **Question (non-blocking):** Do these values need to be passed in as `false`? Going by the liquid component setup, they'll default to `false` if nothing is passed in.

File: src/site/filters/liquid.js [hunk 2]
```diff
@@ -276,18 +276,42 @@ module.exports = function registerFilters() {
...
+  /**
+   * @param {string} phoneNumber a string of a phone number, can be a short number or a long number, however a short number or a number with alphabetic characters will generate a <a> tag instead of a <va-telephone> tag
+   * @param {string} attributes a string of attributes like "not-clickable" or "tty" or "sms" or some combination space separated
+   * @returns string of html that is either a <va-telephone> tag or an <a> tag
+   */
+  liquid.filters.processPhoneToVaTelephoneOrFallback = (
+    phoneNumber = '',
+    attributes = '',
+    describedBy = '',
+  ) => {
```
Comment (maxx1128) [inline src/site/filters/liquid.js:299]: **Suggestion:** It looks like the instances of this that are used have no `attributes,` but some do have `describedBy.` The code would be cleaner to make `describedBy` the second argument and `attributes` the third. That way `describedBy` could just be passed in as needed, and `attributes` wouldn't need to be passed in as an empty string in any usage.

**Review by maxx1128 — CHANGES_REQUESTED** [review]
This is looking good overall, just a couple small adjustments.

File: src/site/paragraphs/staff_profile.drupal.liquid [hunk 3]
```diff
@@ -53,8 +53,7 @@
         vads-u-font-weight--normal
         vads-u-margin--0
         vads-u-margin-bottom--1">
-          <b>Phone:</b> <a
-              href="tel:{{ bio.fieldPhoneNumber }}">{{ bio.fieldPhoneNumber }}</a>
+          <b>Phone:</b> {{ bio.fieldPhoneNumber | processPhoneToVaTelephoneOrFallback :  '', '' }}
```
[the remark this answers is not in the document]
Comment (eselkin) [reply src/site/paragraphs/staff_profile.drupal.liquid:56]: Literally no idea. Strong is more modern, maybe someone had a lapse of like 10 years.

File: src/site/filters/liquid.js (hunk 2 above)
Comment (eselkin) [reply src/site/filters/liquid.js:299]: There are a couple with attributes

File: src/site/components/phone.drupal.liquid (hunk 1 above)
Comment (eselkin) [reply src/site/components/phone.drupal.liquid:13]: I expected that, but for some reason it triggered something when they weren't passed in.

Comment (eselkin) [reply src/site/components/phone.drupal.liquid:13]: I can try again, but it was odd.

File: src/site/paragraphs/staff_profile.drupal.liquid (hunk 3 above)
[the remark this answers is not in the document]
Comment (eselkin) [reply src/site/paragraphs/staff_profile.drupal.liquid:56]: I'll change to strong

[the remark this answers is not in the document]
Comment (eselkin) [reply src/site/paragraphs/staff_profile.drupal.liquid:56]: Changed

File: src/site/components/phone-number.drupal.liquid [hunk 4]
```diff
@@ -13,7 +13,7 @@
     <va-telephone
       contact="{{ partialPhoneNumber | removeDashes }}"
       extension="{{ partialPhoneExtension | default: '' }}"
-      message-aria-describedby="{{ phoneLabel | default: 'Phone' }}"
+      {% if !fax %}message-aria-describedby="{{ phoneLabel | default: 'Phone' }}"{% endif  %}
```
[the remark this answers is not in the document]
Comment (eselkin) [reply src/site/components/phone-number.drupal.liquid]: Changed

File: src/site/filters/liquid.js (hunk 2 above)
Comment (eselkin) [reply src/site/filters/liquid.js:299]: But definitely more do have describedBy

Comment (eselkin) [reply src/site/filters/liquid.js:299]: I guess there's only one that uses attributes now, but it doesn't use describedBy, which makes it equally challenging... But if you don't pass arguments to liquid filters it takes the object of the page as an additional argument which is always present and then you wind up with object values in the contents of filter output.

Comment (maxx1128) [reply src/site/filters/liquid.js:299]: If the page object gets passed in and complicates things, then it's probably best to leave them as-is.

**Review by laflannery — APPROVED** [review]
Reviewed and confirmed all use cases, everything looks good! [Link to QA mural](https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1714566731945/d4e3c5cc3f2b5f19115c9494791c19174de0a80c?sender=u10a597890f85398daa3c4805) for reference

File: src/site/components/phone.drupal.liquid (hunk 1 above)
Comment (eselkin) [reply src/site/components/phone.drupal.liquid]: I removed redundant `= false` assignments. It should be good. Just testing to make sure

**Review by maxx1128 — APPROVED** [review]

**Review by pjhill — APPROVED** [review]

**Review by randimays — APPROVED** [review]
LGTM, nice work!

**Review by jtmst — APPROVED** [review]

**Pull request merged**